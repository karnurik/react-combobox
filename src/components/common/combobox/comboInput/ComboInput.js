import React, { Component } from 'react';
import ComboOptions from '../comboOptions/ComboOptions';
import './comboInput.css';

class ComboInput extends Component {
    constructor(props) {
    super(props);
    this.state = {
        searchValue  : '',
        filteredData : this.props.data
      };
    }

    /* 
      * @description - Set searched value to state variable named searchValue
      * @param event  
      * @return na
    */
    handleSearch = e => {
        const { value } = e.target;
        this.setState({searchValue: value.toLowerCase()},
            _ =>  this.filterOptions() );
    }

    /* 
      * @description - Filters the given data according to the search value
      * @param na  
      * @return na
    */
    filterOptions = _ => {
        let filteredData = this.props.data;
        filteredData = this.props.data.filter( item => item.name.toLowerCase().indexOf(this.state.searchValue) > -1 );
        this.setState({filteredData: filteredData})
    }


    render() {
        const placeholder = this.props.searchPlaceholder ? this.props.searchPlaceholder : 'Search' ;

        return (
           <div className='combo-list-container'>
                <input
                    type='text'
                    name='search'
                    className='form-input'
                    onChange={this.handleSearch}
                    value={this.state.searchValue}
                    placeholder={placeholder}
                    autoComplete='off'
                    autoFocus
                />
                <div className='combo-list'>
                    {this.state.filteredData.map(elem => 
                        <ComboOptions 
                            key={elem.id} elem={elem} 
                            selectedOptions={this.props.selectedOptions} 
                            handleSelectedOptions={this.props.handleSelectedOptions} />
                    )}
                </div>
            </div>
        );
    }
}

export default ComboInput;