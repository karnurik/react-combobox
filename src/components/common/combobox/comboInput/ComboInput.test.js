import React from 'react';
import ComboInput from './ComboInput';
import renderer from 'react-test-renderer';

const data = [
  { id: '1', name: 'Candy Eat' },
  { id: '2', name: 'Onions Pick' },
  { id: '3', name: 'Fanny Pick' },
  { id: '4', name: 'Click o Wisp' }
];
const filteredData = [
  { id: '2', name: 'Onions Pick' },
  { id: '3', name: 'Fanny Pick' },
  { id: '4', name: 'Click o Wisp' }
];
const searchValue = 'p';
const selectedOptions = [];

test('Component renders successfully', () => {
  const component = renderer.create(
    <ComboInput
        data={data}
        filteredData={filteredData}
        searchValue={searchValue}
        selectedOptions={selectedOptions}
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});