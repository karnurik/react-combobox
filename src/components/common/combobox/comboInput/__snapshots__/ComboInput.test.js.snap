// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders successfully 1`] = `
<div
  className="combo-list-container"
>
  <input
    autoComplete="off"
    autoFocus={true}
    className="form-input"
    name="search"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <div
    className="combo-list"
  >
    <div
      className="combo-list-item"
      id="1"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="combo-list-text"
      >
        Candy Eat
      </span>
    </div>
    <div
      className="combo-list-item"
      id="2"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="combo-list-text"
      >
        Onions Pick
      </span>
    </div>
    <div
      className="combo-list-item"
      id="3"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="combo-list-text"
      >
        Fanny Pick
      </span>
    </div>
    <div
      className="combo-list-item"
      id="4"
    >
      <input
        checked={false}
        onChange={[Function]}
        type="checkbox"
      />
      <span
        className="combo-list-text"
      >
        Click o Wisp
      </span>
    </div>
  </div>
</div>
`;
